### PR TITLE
[PM-3505][PM-4587] Update Delete Organization and User SPROCs and EF methods

### PR DESCRIPTION
**Pull request by ike-kottlowski**
## Type of change

```
- [x] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

When trying to delete an Organization the SPROC was failing because of active `AuthRequests` associated with the Organization. Objective is to update SPROC to remove `AuthRequests` in the SPROC.

There another issue were users were not able to be deleted if they had `WebAuthnCredential` assocaited to their account. A similar solution to remove `WebAuthnCredential`in the SPROC was used.

The solutions for Organization and User were applied to both Dapper (SPROC) and Entity Framework flows.

### Added Files

* **2023-12-18_00_RemoveAuthRequest_OrganizationDeleteById.sql:** created a migration that updated the SPROC to delete Organizaitons
* **2023-12-20_00_RemovePassKeys_UserDeleteById.sql:** created a migration that updated the SPROC to delete Users

## Updated Files

* **OrganizationRepository.cs:** Added call to remove `AuthRequest`s associated with the organization.
* **UserRepository.cs:** Added call to remove `WebAuthnCredential`s associated with the user.

## Before you submit

- Please check for formatting errors (`dotnet format --verify-no-changes`) (required)
- If making database changes - make sure you also update Entity Framework queries and/or migrations
- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

File: util/Migrator/DbScripts/2023-12-18_00_RemoveAuthRequest_OrganizationDeleteById.sql [hunk 1]
```diff
@@ -0,0 +1,134 @@
+IF OBJECT_ID('[dbo].[Organization_DeleteById]') IS NOT NULL
+BEGIN
+    DROP PROCEDURE [dbo].[Organization_DeleteById]
+END
+GO
+
+CREATE PROCEDURE [dbo].[Organization_DeleteById]
+    @Id UNIQUEIDENTIFIER
+AS
```
Comment (rkac-bw) [inline util/Migrator/DbScripts/2023-12-18_00_RemoveAuthRequest_OrganizationDeleteById.sql:9]: If this runs infrequently may want to add with recompile here also

Comment (ike-kottlowski) [reply util/Migrator/DbScripts/2023-12-18_00_RemoveAuthRequest_OrganizationDeleteById.sql:11]: updated.

**Comment by withinfocus** [comment]
The DB validator is indicating that you also need to make that change in the master copy of the proc.

**Review by rkac-bw — APPROVED** [review]
looks good to me

**Review by r-tome — APPROVED** [review]

**Pull request merged**